src/data/nat/modeq.lean

```diff
@@ -129,6 +129,55 @@ else by rw [← @add_right_cancel_iff _ _ (c * (a / b / c)), mod_add_div, nat.di
   mul_add, ← @add_left_cancel_iff _ _ (a % (b * c) % b), add_left_comm,
   ← add_assoc (a % (b * c) % b), mod_add_div, ← mul_assoc, mod_add_div, mod_mul_right_mod]
 
+lemma add_mod_add_ite {a b c : ℕ} (hc0 : 0 < c) :
```

Comment:
You don't need the positivity of `c` here, as the statement is trivially true for `c = 0` also.

Reply:
`a % 0` is defined to be `a` not `0`, so the hypothesis required.

Reply:
For `c = 0`, the lemma is saying that `a + b + (if true then 0 else 0) = a + b`, which looks true to me.

Reply:
Good point, I'm an idiot.